### PR TITLE
Add NxtSupport::Refinements::Crystalizer

### PR DESCRIPTION
**Pull request by AkihikoITOH**

### CODE REVIEW

File: lib/nxt_support/refinements/crystalizer.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module NxtSupport
+  module Refinements
+    module Crystalizer
```
Comment (rkallensee) [inline lib/nxt_support/refinements/crystalizer.rb:3]: Quick idea (might be not thought through): what about calling it `ArrayCrystalizer` to make it more clear that it refines `Array`?

**Review by rkallensee — APPROVED** [review]
I like it! 👏🏼

**Review by ston1x — APPROVED** [review]

**Comment by nsommer** [comment]
@AkihikoITOH can we merge this?

File: lib/nxt_support/refinements/crystalizer.rb (hunk 1 above)
Comment (AkihikoITOH) [reply lib/nxt_support/refinements/crystalizer.rb:3]: I just realised refining `Enumerable` works with Ruby 3.0. I'll keep the name and switch the behaviour based on ruby version.

**Comment by AkihikoITOH** [comment]
> @AkihikoITOH can we merge this?

I'll quickly add support for Ruby 3.0 and then merge!

**Pull request merged**